packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -1464,7 +1475,15 @@ function reifySelectionDefaults(placement: SubnetSelection): SubnetSelection {
     throw new Error(`Only one of 'subnetType' and 'subnetGroupName' can be supplied`);
   }
 
-  if (placement.subnetType === undefined && placement.subnetGroupName === undefined) {
+  if (placement.subnets !== undefined && placement.subnetType !== undefined) {
```

Comment:
Can you make this `if` compare all 3 parameters at the same time and produce an appropriate error message?

Suggested code:

```ts
const possibleKeys: Array<keyof SubnetSelection> = ['subnets', 'subnetType', 'subnetGroupName'];
const provided = keys.filter(key => placement[key] !== undefined);

if (provided.length > 1) {
  throw new Error(`'${provided}' specified together in subnet selection, specify only one`);
}
```